### PR TITLE
The removeIndex() is not called when current record is null.

### PR DESCRIPTION
**Pull request by jerrinot**
Fixes findbugs issues. One of them turns out to be a real bug.

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java [hunk 1]
```diff
@@ -629,7 +629,6 @@ public boolean delete(Data key) {
 
         final Record record = getRecordOrNull(key, now, false);
         if (record == null) {
-            removeIndex(record);
```
Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java]: this must have been some left-over not covered by test. 
The `removeIndex()` really does not make sense when the record is not stored locally. This change makes it aligned with the `remove()` method.

**Comment by ahmetmircik** [comment]
:+1:

File: hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java (hunk 1 above)
Comment (serkan-ozal) [inline hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java]: @jerrinot you seems right but maybe there was another intention. @ahmetmircik can you have a look?

**Comment by serkan-ozal** [comment]
:+1:

File: hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java (hunk 1 above)
Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java]: @serkan-ozal: I have already discussed this with @ahmetmircik over slack.

Comment (serkan-ozal) [inline hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java]: ok, lets merge it

**Pull request merged**